drivers/include/periph/pm.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_pm Power Management
+ * @ingroup     drivers_periph
+ * @brief       The kernels power management interface
+ * @{
+ *
+ * This simple power management interface is based on the following assumptions:
+ *
+ * - CPUs define up to 4 power modes (from zero, the lowest power mode, to
+ *   PM_NUM_MODES-1, the highest)
+ * - there is an implicit extra idle mode (which has the number PM_NUM_MODES)
+ * - individual power modes can be blocked/unblocked, e.g., by peripherals
+ * - if a mode is blocked, so are implicitly all lower modes
```

Comment:
If `blocker.val_u32=0` the new loop will try to set UINTN_MAX as mode
If you change 
```c
    while (mode--) {
```
to 
```c
    while (mode) {
        mode--;
```
The correct mode (0) will be selected